packages/core/src/generator/section-generator.adapter.ts

```diff
@@ -32,6 +32,7 @@ export interface SectionGenerationPayload<TManifest> {
   formatterAdapter: FormatterAdapter;
   manifest: TManifest;
   repositoryProvider: RepositoryProvider;
+  destination?: string; // Optional destination path for relative path calculations
```

Comment:
```suggestion
  destination: string;
```

Destination must be always provided, to keep the code Clean